src/main/java/org/broadinstitute/hellbender/metrics/InsertSizeMetricsCollector.java

```diff
@@ -53,135 +69,154 @@ protected InsertSizeMetricsCollectorArgs makeArg(SAMRecord samRecord, ReferenceS
         return new InsertSizeMetricsCollectorArgs(insertSize, orientation);
     }
 
-    /** Make an InsertSizeCollector with the given arguments */
+    // Make an PerUnitInsertSizeMetricsCollector with the given arguments
     @Override
-    protected PerUnitMetricCollector<InsertSizeMetrics, Integer, InsertSizeMetricsCollectorArgs> makeChildCollector(final String sample, final String library, final String readGroup) {
-        return new PerUnitInsertSizeMetricsCollector(sample, library, readGroup);
+    protected PerUnitInsertSizeMetricsCollector makeChildCollector(
+            final String sample,
+            final String library,
+            final String readGroup) {
+        return new PerUnitInsertSizeMetricsCollector(
+                sample,
+                library,
+                readGroup,
+                inputArgs.minimumPct,
+                inputArgs.maxMADTolerance,
+                inputArgs.histogramWidth);
     }
 
+    /**
+     * Process a single SAMRecord. The record is assumed to have already been filtered
+     * by the caller using the InsertSizeMetricsReadFilter.
+     * @param record SAMRecord
+     * @param refSeq
+     */
     @Override
     public void acceptRecord(final SAMRecord record, final ReferenceSequence refSeq) {
-        if (!record.getReadPairedFlag() ||
-                record.getReadUnmappedFlag() ||
-                record.getMateUnmappedFlag() ||
-                record.getFirstOfPairFlag() ||
-                record.isSecondaryOrSupplementary() ||
-                record.getDuplicateReadFlag() ||
-                record.getInferredInsertSize() == 0) {
-            return;
-        }
-
         super.acceptRecord(record, refSeq);
```

Comment:
if all this method does is call super, then it's useless and we could delete it from here.